packages/apollo-gateway/src/index.ts

```diff
@@ -385,6 +411,33 @@ export class ApolloGateway implements GraphQLService {
     }
   }
 
+  private notifySchemaListeners(schema?: GraphQLSchema) {
+    try {
+      this.onSchemaChangeListeners.forEach(listener => listener(schema!));
+    } catch (e) {
+      this.logger.error(
+        "An error was thrown from an 'onSchemaChange' listener. " +
+        "The schema will still update: " + (e && e.message || e));
+    }
+  }
+
+  // This can be used without an argument in order to perform an ad-hoc health check
+  // of the downstream services like so:
+  // try {
+  //   await gateway.performServiceHealthChecks();
+  // } catch(e) {
+  //   /* your error handling here */
+  // }
+  protected performServiceHealthChecks(serviceMap: DataSourceMap = this.serviceMap) {
+    return Promise.all(
+      Object.entries(serviceMap).map(([name, { dataSource }]) =>
+        dataSource
+          .process({ request: { query: HEALTH_CHECK_QUERY }, context: {} })
```

Comment:
I suppose we'll gain retry-ability on this if we introduce support for retries to `RemoteGraphQLDataSource.prototype.process`.  Not asking for that change, just noting out-loud. :wink:

Reply:
Yes!